### PR TITLE
Makes .js files modules

### PR DESCRIPTION
**Pull request by jaysaleh**
This PR makes all `.js `files modules such that functions can be imported/exported to other `.js` files, eliminating the need for `window.`

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -12,6 +13,6 @@
     "no-multi-spaces": "off",
     "require-jsdoc": "off",
     "valid-jsdoc": "off",
-    "no-unused-vars": ["error", { "vars": "local", "args": "after-used", "ignoreRestSiblings": false }]
+    "no-unused-vars": ["off", { "vars": "local", "args": "after-used", "ignoreRestSiblings": false }]
```
Comment (n-wach) [inline .eslintrc.json:16]: Can this be switched back to error?

File: src/main/webapp/view/view.js [hunk 2]
```diff
@@ -12,6 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import {intializeDiscussion} from '../view/discussion/discussion.js';
+
```
Comment (n-wach) [inline src/main/webapp/view/view.js:16]: Can the newlines between imports be removed? Or was that the linter

Comment (jaysaleh) [reply src/main/webapp/view/view.js:16]: That was the linter

File: src/main/webapp/view/discussion/discussion.js [hunk 3]
```diff
@@ -39,6 +39,8 @@ const SELECTOR_REPLY_TEXTAREA = '#reply-textarea';
 // 10 seconds.
 const TIME_TOLERANCE_MS = 10000;
 
+window.postNewComment = postNewComment;
```
Comment (f-camp) [inline src/main/webapp/view/discussion/discussion.js:42]: Should we be able to eliminate window. here?

Comment (n-wach) [reply src/main/webapp/view/discussion/discussion.js:42]: Unfortunately not at the moment, because it is used in the HTML

**Pull request merged**